pkg/resources/machinecontroller/deployment.go

```diff
@@ -281,6 +285,10 @@ func getFlags(clusterDNSIP string, nodeSettings *kubermaticv1.NodeSettings, cri
 			flags = append(flags, "-node-pause-image", nodeSettings.PauseImage)
 		}
 
+		if isExternal {
+			flags = append(flags, "-node-external-cloud-provider", "true")
```

Comment:
Please remove the "true", it would cause parsing errors in Go's flag handling. @xmudrii and me just recently discovered a similar bug in our machine-controller code (see `-node-csr-approver` flag above, which is also incorrect, but soon fixed in #7461)

Reply:
done